### PR TITLE
Eliminate hack

### PR DESCRIPTION
**Pull request by etuleu**
Thanks to @garymm for uncovering and helping figure out the need for this sleep hack in startServer.

A few things were going on:
1. The waitgroup in startServer was being passed by value which meant it was getting copied and so the add method was adding to a different object.
2. The NewServer method was not calling transport open which binds the socket to an actual port, which I guess takes a bit of time. By calling open in the create function we can eliminate the need to wait later.

### CODE REVIEW

File: infra/odpw/example.cfg [hunk 1]
```diff
@@ -15,7 +15,7 @@ targetScore=5.2
 scoreTolerance=0.1
 targetCycles=100.2
 cycleTolerance=0.1
-buildOperation="go build"
```
Comment (garymm) [inline infra/odpw/example.cfg:18]: can you describe this in the PR description too, or move it to a separate PR?

File: models/integrated/simple_supervised/simple_agent_test.go [hunk 2]
```diff
@@ -25,6 +25,13 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
+func startServer(server *thrift.TSimpleServer, wg *sync.WaitGroup) {
+	wg.Add(1)
+	defer wg.Done()
+	log.Info("PROTOCOL: World server listening on: ", WorldAddr)
```
Comment (garymm) [inline models/integrated/simple_supervised/simple_agent_test.go:31]: Not that it really matters, but for readability I'd move this below the call to `Serve()`, since at least the goroutine exists at that point.

File: network/network.go [hunk 3]
```diff
@@ -5,40 +5,46 @@ import (
...
-	protocolFactory := thrift.NewTBinaryProtocolFactoryConf(nil)
-	return thrift.NewTSimpleServer4(processor, transport, transportFactory, protocolFactory)
+	return thrift.NewTSimpleServer2(processor, transport), nil
 }
 
-func NewAgentClient(addr string) *env.AgentClient {
+func NewAgentClient(addr string) (*env.AgentClient, error) {
 	transportFactory := thrift.NewTBufferedTransportFactory(8192)
 	transportSocket := thrift.NewTSocketConf(addr, nil)
 	transport, _ := transportFactory.GetTransport(transportSocket)
```
Comment (garymm) [inline network/network.go:29]: don't ignore the error here, since we can now surface it easily to the caller

File: network/network.go [hunk 4]
```diff
@@ -5,40 +5,46 @@ import (
...
+		return nil, err
+	}
+	return env.NewAgentClient(thrift.NewTStandardClient(iprot, oprot)), nil
 }
 
-func NewWorldClient(addr string) *env.WorldClient {
+func NewWorldClient(addr string) (*env.WorldClient, error) {
 	transportFactory := thrift.NewTBufferedTransportFactory(8192)
 	transportSocket := thrift.NewTSocketConf(addr, nil)
 	transport, _ := transportFactory.GetTransport(transportSocket)
```
Comment (garymm) [inline network/network.go:42]: ditto

File: network/network.go [hunk 5]
```diff
@@ -5,40 +5,46 @@ import (
...
+	return thrift.NewTSimpleServer2(processor, transport), nil
 }
 
-func NewAgentClient(addr string) *env.AgentClient {
+func NewAgentClient(addr string) (*env.AgentClient, error) {
 	transportFactory := thrift.NewTBufferedTransportFactory(8192)
 	transportSocket := thrift.NewTSocketConf(addr, nil)
 	transport, _ := transportFactory.GetTransport(transportSocket)
 	protocolFactory := thrift.NewTBinaryProtocolFactoryConf(nil)
 	iprot := protocolFactory.GetProtocol(transport)
```
Comment (garymm) [inline network/network.go:31]: I know this is just from example code, but I think it's slightly more readable to use camelCase here:
`iProt` and `oProt` instead of `iprot` and `oprot`

File: network/network.go [hunk 6]
```diff
@@ -5,40 +5,46 @@ import (
...
+	}
 	processor := env.NewWorldProcessor(handler)
-	protocolFactory := thrift.NewTBinaryProtocolFactoryConf(nil)
-	return thrift.NewTSimpleServer4(processor, transport, transportFactory, protocolFactory)
+	return thrift.NewTSimpleServer2(processor, transport), nil
 }
 
-func NewAgentClient(addr string) *env.AgentClient {
+func NewAgentClient(addr string) (*env.AgentClient, error) {
 	transportFactory := thrift.NewTBufferedTransportFactory(8192)
```
Comment (garymm) [inline network/network.go:27]: I don't think we should keep buffering here but not on the server.
I guess it's a tradeoff between throughput and latency. I think either is fine until we actually try it and see which one works better, but for now I think it's best to be consistent.

**Review by garymm — APPROVED** [review]

File: infra/odpw/example.cfg (hunk 1 above)
Comment (etuleu) [reply infra/odpw/example.cfg:18]: Oh actually this file is not needed anymore I think. I'll revert this change.
